README.md
I just ask to mention
its mod creator and source
if you want to use it for you.

Je demande juste de mentionner
son créateur et sa source de mod
si vous voulez l'utiliser pour vous.

changelog.txt
---------------------------------------------------------------------------------------------------
Version: 0.0.4
Date: 07.03.2021
  Bugfixes:
    - Fixed incorrect default value
---------------------------------------------------------------------------------------------------
Version: 0.0.3
Date: 07.03.2021
  Bugfixes:
    - Fixed an error in the command that prevents it from working in multiplayer
---------------------------------------------------------------------------------------------------
Version: 0.0.2
Date: 07.03.2021
  Changes:
    - Added Always days
    - Added Command for Multiplayer
  Bugfixes:
    - Missing equipment icons not showing up correctly.
---------------------------------------------------------------------------------------------------
Version: 0.0.1
Date: 07.03.2021
  Changes:
    - Added basic operation
    - Added Life point
    - Added Interaction distance with the environment
    - Added Interaction distance with objects on the ground
    - Added Deconstruction / destruction speed multiplier
    - Added Manufacturing speed multiplier
    - Added Number of locations in the inventory
    - Added Number of places in the trash
    - Added Increased efficiency of drilling rigs
    - Added Increased laboratory speed
    - Added Increased laboratory efficiency
    - Added Increased movement speed
---------------------------------------------------------------------------------------------------

control.lua
--
require "script.cmds"
--
script.on_init(function ()
	apply_unrealplayer_settings()
end)
--
script.on_configuration_changed(function (data)
	apply_unrealplayer_settings()
end)
--
script.on_event(defines.events.on_runtime_mod_setting_changed,function ()
	apply_unrealplayer_settings()
end)
--
function apply_unrealplayer_settings()
	local settings = settings.global
	game.surfaces[1].always_day = settings["mj_day"]["value"]
	game.forces["player"].character_health_bonus = settings["mj_health"]["value"]
	game.forces["player"].character_reach_distance_bonus = settings["mj_distance"]["value"]
	game.forces["player"].character_resource_reach_distance_bonus = settings["mj_distance"]["value"]
	game.forces["player"].character_build_distance_bonus = settings["mj_distance"]["value"]
	game.forces["player"].character_item_drop_distance_bonus = settings["mj_distance"]["value"]
	game.forces["player"].character_item_pickup_distance_bonus = settings["mj_pickup"]["value"]
	game.forces["player"].character_loot_pickup_distance_bonus = settings["mj_pickup"]["value"]
	game.forces["player"].manual_mining_speed_modifier = settings["mj_mining"]["value"]
	game.forces["player"].manual_crafting_speed_modifier = settings["mj_crafting"]["value"]
	game.forces["player"].character_inventory_slots_bonus = settings["mj_inventory"]["value"]
	game.forces["player"].character_trash_slot_count = settings["mj_trash"]["value"]
	game.forces["player"].mining_drill_productivity_bonus = settings["mj_drill"]["value"]
	game.forces["player"].laboratory_speed_modifier = settings["mj_laboratory_speed"]["value"]
	game.forces["player"].laboratory_productivity_bonus = settings["mj_laboratory_productivity"]["value"]
	game.forces["player"].character_running_speed_modifier = settings["mj_running"]["value"]
end
--

settings.lua
data:extend({
   {
      type="bool-setting",
      name = "mj_day",
      setting_type = "runtime-global",
      default_value = false,
      per_user = false,
      order = "aa"
   },
   {
      type = "double-setting",
      name = "mj_health",
      setting_type = "runtime-global",
      default_value = 250,
      maximum_value = 9999999,
      minimum_value = 250,
      per_user = false,
      order = "ab"
   },
   {
      type = "int-setting",
      name = "mj_distance",
      setting_type = "runtime-global",
      default_value = 6,
      maximum_value = 9999999,
      minimum_value = 6,
      per_user = false,
      order = "ac"
   },
   {
      type = "double-setting",
      name = "mj_pickup",
      setting_type = "runtime-global",
      default_value = 6,
      maximum_value = 30,
      minimum_value = 6,
      per_user = false,
      order = "ad"
   },
   {
      type = "double-setting",
      name = "mj_mining",
      setting_type = "runtime-global",
      default_value = 0,
      maximum_value = 9999999,
      minimum_value = 0,
      per_user = false,
      order = "ae"
   },
   {
      type = "double-setting",
      name = "mj_crafting",
      setting_type = "runtime-global",
      default_value = 0,
      maximum_value = 9999999,
      minimum_value = 0,
      per_user = false,
      order = "af"
   },
   {
      type = "int-setting",
      name = "mj_inventory",
      setting_type = "runtime-global",
      default_value = 0,
      maximum_value = 9999999,
      minimum_value = 0,
      per_user = false,
      order = "ah"
   },
   {
      type = "int-setting",
      name = "mj_trash",
      setting_type = "runtime-global",
      default_value = 30,
      maximum_value = 9999999,
      minimum_value = 30,
      per_user = false,
      order = "ai"
   },
   {
      type = "double-setting",
      name = "mj_running",
      setting_type = "runtime-global",
      default_value = 0,
      maximum_value = 9999999,
      minimum_value = 0,
      per_user = false,
      order = "ag"
   },
   {
      type = "double-setting",
      name = "mj_drill",
      setting_type = "runtime-global",
      default_value = 0,
      maximum_value = 9999999,
      minimum_value = 0,
      per_user = false,
      order = "ba"
   },
   {
      type = "double-setting",
      name = "mj_laboratory_speed",
      setting_type = "runtime-global",
      default_value = 0,
      maximum_value = 9999999,
      minimum_value = 0,
      per_user = false,
      order = "bb"
   },
   {
      type = "double-setting",
      name = "mj_laboratory_productivity",
      setting_type = "runtime-global",
      default_value = 0,
      maximum_value = 9999999,
      minimum_value = 0,
      per_user = false,
      order = "bc"
   }
})

script/cmds.lua
commands.add_command("up", "[UnrealPlayer] Used to control metrics in Unreal Player\n/up [list | day true-false | health 250-9999999 | distance 6-9999999 | pickup 6-30 | mining 0-9999999 | crafting 0-9999999 | inventory 0-9999999 | trash 30-9999999 | drill 0-9999999 | laboratory_speed 0-9999999 | laboratory_productivity 0-9999999 | running 0-9999999] ", function(param)
  player = game.players[param.player_index]
  if not player then return end
  if not player.admin then
    player.print("[UnrealPlayer] Commands are only available to admins")
    return
  end
  argsArr = split(param.parameter, " ")
  if argsArr[1] == "day" then
    game.surfaces[1].always_day = argsArr[2]
    player.print("[UnrealPlayer] Value always day changed!")
  elseif argsArr[1] == "health" then
    game.forces["player"].character_health_bonus = argsArr[2]
    player.print("[UnrealPlayer] Value health changed!")
  elseif argsArr[1] == "distance" then
    game.forces["player"].character_reach_distance_bonus = argsArr[2]
    game.forces["player"].character_resource_reach_distance_bonus = argsArr[2]
    game.forces["player"].character_build_distance_bonus = argsArr[2]
    game.forces["player"].character_item_drop_distance_bonus = argsArr[2]
    player.print("[UnrealPlayer] Value distance changed!")
  elseif argsArr[1] == "pickup" then
    game.forces["player"].character_item_pickup_distance_bonus = argsArr[2]
    game.forces["player"].character_loot_pickup_distance_bonus = argsArr[2]
    player.print("[UnrealPlayer] Value pickup changed!")
  elseif argsArr[1] == "mining" then
    game.forces["player"].manual_mining_speed_modifier = argsArr[2]
    player.print("[UnrealPlayer] Value mining changed!")
  elseif argsArr[1] == "crafting" then
    game.forces["player"].manual_crafting_speed_modifier = argsArr[2]
    player.print("[UnrealPlayer] Value crafting changed!")
  elseif argsArr[1] == "inventory" then
    game.forces["player"].character_inventory_slots_bonus = argsArr[2]
    player.print("[UnrealPlayer] Value inventory changed!")
  elseif argsArr[1] == "trash" then
    game.forces["player"].character_trash_slot_count = argsArr[2]
    player.print("[UnrealPlayer] Value trash changed!")
  elseif argsArr[1] == "drill" then
    game.forces["player"].mining_drill_productivity_bonus = argsArr[2]
    player.print("[UnrealPlayer] Value drill changed!")
  elseif argsArr[1] == "laboratory_speed" then
    game.forces["player"].laboratory_speed_modifier = argsArr[2]
    player.print("[UnrealPlayer] Value laboratory_speed changed!")
  elseif argsArr[1] == "laboratory_productivity" then
    game.forces["player"].laboratory_productivity_bonus = argsArr[2]
    player.print("[UnrealPlayer] Value laboratory_productivity changed!")
  elseif argsArr[1] == "running" then
    game.forces["player"].character_running_speed_modifier = argsArr[2]
    player.print("[UnrealPlayer] Value running changed!")
  elseif argsArr[1] == "list" then
    player.print("[UnrealPlayer] /up [list | day true-false | health 250-9999999 | distance 6-9999999 | pickup 6-30 | mining 0-9999999 | crafting 0-9999999 | inventory 0-9999999 | trash 30-9999999 | drill 0-9999999 | laboratory_speed 0-9999999 | laboratory_productivity 0-9999999 | running 0-9999999]")
  else
    player.print("[UnrealPlayer] Try list, For example:\n/up list")
  end
end)

function split(str, character)
  result = {}

  index = 1

  if str == nil then return result end
  if character == nil then return result end

  for s in string.gmatch(str, "[^"..character.."]+") do
    result[index] = s
    index = index + 1
  end

  return result
end